### PR TITLE
Jobseekers can withdraw an application

### PR DESCRIPTION
**Pull request by cpjmcquillan**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-1953

## Changes in this PR
- Jobseekers can withdraw submitted or shortlisted job applications

### CODE REVIEW

File: app/controllers/jobseekers/job_applications_controller.rb [hunk 1]
```diff
@@ -43,6 +43,27 @@ def destroy
...
+    raise ActionController::RoutingError, "Cannot withdraw non-submitted or non-shortlisted application" unless
+      job_application.status.in?(%w[shortlisted submitted])
+  end
+
+  def withdraw
+    raise ActionController::RoutingError, "Cannot withdraw non-submitted or non-shortlisted application" unless
+      job_application.status.in?(%w[shortlisted submitted])
+
+    if params[:commit] == t("buttons.cancel")
+      return redirect_to jobseekers_job_applications_path if URI(params[:origin]).path == jobseekers_job_applications_path
```
Comment (alexbowen) [inline app/controllers/jobseekers/job_applications_controller.rb:56]: there is a line of code after the return?

Comment (cpjmcquillan) [reply app/controllers/jobseekers/job_applications_controller.rb:56]: there's an
```ruby
if
```
after it

File: spec/system/jobseekers_can_withdraw_a_job_application_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  end
+
+  it "submits the form, renders error, then withdraws the application" do
+    click_on I18n.t("buttons.withdraw_application")
+
+    expect(page).to have_content("There is a problem")
+
+    choose "jobseekers-job-application-withdraw-form-withdraw-reason-other-field"
+
+    expect { click_on I18n.t("buttons.withdraw_application") }
```
Comment (cesidio) [inline spec/system/jobseekers_can_withdraw_a_job_application_spec.rb:22]: Not entirely sure, but if we test record creation in request specs, we can probably test only the interface in system spec, things that are on the screen.

**Review by cesidio — APPROVED** [review]

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-3156.london.cloudapps.digital>

File: spec/system/jobseekers_can_withdraw_a_job_application_spec.rb (hunk 2 above)
Comment (cpjmcquillan) [reply spec/system/jobseekers_can_withdraw_a_job_application_spec.rb:22]: Yeah I'm with you on that actually. 

I'd like to keep system specs as clean and simple as possible across the whole application 👍

**Pull request merged**